Azure/azure-service-operator
# Add check to only provision SQL User once

Closes #843 

**What this PR does / why we need it**:
SQL User should only go through the user create flow once, and stop once provisioned.

**Special notes for your reviewer**:
To test:
Create SQL Server + DB
Create Key Vault
Create KV
Create DB User (with Key Vault to store secrets)
Check secret in Key Vault. Restart debugger or operator. Check secret and ensure that no new versions have not been added.

**How does this PR make you feel**:
![gif](https://media.giphy.com/media/fTnmUs24FN1ifdqciN/giphy.gif)

**If applicable**:
